docs/tutorials/08-tutorial-sug-roc-notleap.ipynb

```diff
@@ -0,0 +1,981 @@
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "a7545a70-e6b0-43ba-a8f7-a2d02d48eec6",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 119/119 [10:41<00:00,  5.39s/it]\n"
+     ]
+    }
+   ],
+   "source": [
+    "import shutil\n",
+    "\n",
+    "from paprika.restraints.openmm import apply_dat_restraint, apply_positional_restraints\n",
+    "from paprika.utils import index_from_mask\n",
+    "from tqdm import tqdm\n",
+    "\n",
+    "for window in window_list:\n",
+    "    dir = f\"{workdir}/{window}\"\n",
+    "    if not os.path.isdir(dir):\n",
+    "        os.makedirs(dir)\n",
+    "\n",
+    "    if window[0] == \"a\":\n",
+    "        shutil.copy(\n",
+    "            f\"{compdir}/aligned_solvated_dummy_structure.pdb\",\n",
+    "            f\"{dir}/aligned_solvated_dummy_structure.pdb\",\n",
+    "        )\n",
+    "\n",
+    "pdb = PDBFile(f\"{compdir}/aligned_solvated_dummy_structure.pdb\")\n",
+    "old_positions = pdb.positions\n",
+    "step_size = 0.25\n",
+    "assert (\n",
+    "    step_size <= distance_step\n",
+    "), \"The step size of the manual pulling must be <= the step size of the automatic pulling step size!\"\n",
+    "for offset in tqdm(\n",
+    "    np.arange(min(pull_distances), max(pull_distances) + step_size, step_size)[:]\n",
+    "):\n",
+    "    offset = round(offset, 2)\n",
+    "    # assert offset < initial_distance + 0.001\n",
+    "\n",
+    "    with open(f\"{compdir}/solvated_dummy_system.xml\", \"r\") as f:\n",
+    "        system = XmlSerializer.deserialize(f.read())\n",
+    "\n",
+    "    # Apply dummy static restraints\n",
+    "    apply_positional_restraints(\n",
+    "        f\"{compdir}/aligned_solvated_dummy_structure.pdb\", system, force_group=15\n",
+    "    )\n",
+    "\n",
+    "    # Apply host static restraints\n",
+    "    for restraint in static_restraints:\n",
+    "        apply_dat_restraint(system, restraint, \"pull\", 1, force_group=10)\n",
+    "\n",
+    "    bond_energy_expression = \"k_bond * (r - r_0)^2;\"\n",
+    "    bond_restraint = CustomBondForce(bond_energy_expression)\n",
+    "    bond_restraint.addPerBondParameter(\"k_bond\")\n",
+    "    bond_restraint.addPerBondParameter(\"r_0\")\n",
+    "    bond_restraint.addBond(\n",
+    "        index_from_mask(f\"{compdir}/aligned_solvated_dummy_structure.pdb\", D1, False)[\n",
+    "            0\n",
+    "        ],\n",
+    "        index_from_mask(f\"{compdir}/aligned_solvated_dummy_structure.pdb\", G1, False)[\n",
+    "            0\n",
+    "        ],\n",
+    "        [\n",
+    "            500 * kilocalorie / angstrom**2 / mole,\n",
+    "            (offset) * angstrom,\n",
+    "        ],\n",
+    "    )\n",
+    "    system.addForce(bond_restraint)\n",
+    "\n",
```

Comment:
Not critical, but you could probably simplify the addition of these restraints with the function `apply_dat_restraints`

Reply:
Using OpenMM Forces directly make it a little more verbose but I find it less confusing because the two pullings are separated: automatic pulling is left up to pAPRika and the manual pulling is done directly in OpenMM. If I were to change it to pAPRika restraints do you think it would be easier for users to follow the tutorial?